### PR TITLE
markers_data

### PR DESCRIPTION
**Pull request by AurelienRenou**
Added choice for marker data type, user can use directly a table with marker position, a c3d file or just use the path to the c3d file.

#11

### CODE REVIEW

File: static_inverse_kinematics.py [hunk 1]
```diff
@@ -56,28 +57,42 @@ class StaticInverseKinematics:
...
+            self.c3d = marker_data
+            self.xp_markers = self._get_marker_trajectories()
+            self.nb_frames = self.c3d["parameters"]["POINT"]["FRAMES"]["value"][0]
+
+        elif type(marker_data) is np.ndarray:
+            if marker_data.ndim == 3 and marker_data.shape[0] <= 3 and marker_data.shape[1] == self.nb_markers:
+                self.xp_markers = marker_data
+                self.nb_frames = marker_data.shape[2]
+            else:
+                raise Exception(
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: Value Error

File: static_inverse_kinematics.py [hunk 2]
```diff
@@ -56,28 +57,42 @@ class StaticInverseKinematics:
...
+                    "The markers data hasn't dimension expected. dim should be (nb_dim, nb_marker, nb_frame).")
+        else:
+            raise Exception(
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: ValueError

File: static_inverse_kinematics.py [hunk 3]
```diff
@@ -56,28 +57,42 @@ class StaticInverseKinematics:
...
         self.marker_names = [
             self.biorbd_model.markerNames()[i].to_string() for i in range(len(self.biorbd_model.markerNames()))
         ]
-        self.xp_markers = self._get_marker_trajectories()
-        self.nb_q = self.biorbd_model.nbQ()
-        self.nb_frames = self.c3d["parameters"]["POINT"]["FRAMES"]["value"][0]
         self.nb_markers = self.biorbd_model.nbMarkers()
 
-        self._get_idx_to_remove()
+        if type(marker_data) is str:
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: use isinstance instead of type() + is

File: static_inverse_kinematics.py [hunk 4]
```diff
@@ -56,28 +57,42 @@ class StaticInverseKinematics:
...
+            self.c3d_path_file = marker_data
+            self.c3d = c3d(self.c3d_path_file)
+            self.xp_markers = self._get_marker_trajectories()
+            self.nb_frames = self.c3d["parameters"]["POINT"]["FRAMES"]["value"][0]
+
+        elif type(marker_data) is c3d:
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: same

File: static_inverse_kinematics.py [hunk 5]
```diff
@@ -56,28 +57,42 @@ class StaticInverseKinematics:
...
+            self.c3d = marker_data
+            self.xp_markers = self._get_marker_trajectories()
+            self.nb_frames = self.c3d["parameters"]["POINT"]["FRAMES"]["value"][0]
+
+        elif type(marker_data) is np.ndarray:
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: same

File: static_inverse_kinematics.py [hunk 6]
```diff
@@ -56,28 +57,42 @@ class StaticInverseKinematics:
...
+            if marker_data.ndim == 3 and marker_data.shape[0] <= 3 and marker_data.shape[1] == self.nb_markers:
+                self.xp_markers = marker_data
+                self.nb_frames = marker_data.shape[2]
+            else:
+                raise Exception(
+                    "The markers data hasn't dimension expected. dim should be (nb_dim, nb_marker, nb_frame).")
+        else:
+            raise Exception(
+                "The markers data hasn't type expected. It should be a str, a ezc3d.c3d or a numpy.ndarray.")
```
Comment (Ipuch) [inline static_inverse_kinematics.py]: "The standard inputs are str, an ezc3d.c3d, or a numpy.ndarray"

File: static_inverse_kinematics.py (hunk 1 above)
Comment (Ipuch) [reply static_inverse_kinematics.py]: f"The standard dimension of the NumPy array should be (nb_dim, nb_marker, nb_frame)"

File: static_inverse_kinematics.py (hunk 4 above)
Comment (Ipuch) [reply static_inverse_kinematics.py]: Plz fill others attributes with None such as ```python self.c3d_path_file =None```

**Review by Ipuch — COMMENTED** [review]
Let's do this.  :+1:

File: static_inverse_kinematics.py (hunk 6 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py]: done

File: static_inverse_kinematics.py (hunk 1 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py]: done

File: static_inverse_kinematics.py (hunk 2 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py]: done

File: static_inverse_kinematics.py (hunk 3 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py]: done

File: static_inverse_kinematics.py (hunk 4 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py]: done

File: static_inverse_kinematics.py (hunk 5 above)
Comment (AurelienRenou) [reply static_inverse_kinematics.py]: done

**Comment by Ipuch** [comment]
LGTM

**Pull request merged**